packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -217,124 +180,75 @@ const DataGrid = injectIntl((props) => {
     props.focusFuncRef,
     () => ({
       setFocusedRowCol,
-      getFocusedCell() { return { row: focusedRow, col: focusedCol }; },
+      getFocusedCell() { return { row: focusedRow.current, col: focusedCol.current }; },
     }),
     // eslint-disable-next-line react-hooks/exhaustive-deps
     [focusedCol, focusedRow],
   );
 
-  // -------------------------------------
-  // callback Hooks
-
-  const gridRef = useCallback((node) => {
-    if (!node) {
-      return;
-    }
-
-    grid.current = node;
-
-    const resizeObserver = new ResizeObserver(() => {
-      // Update table height state variable
-      setTableHeight(grid.current.offsetHeight - 1);
-    });
-
-    // Register resize observer to detect size changes
-    resizeObserver.observe(node);
-  }, []);
-
   // -------------------------------------
   // useEffect Hooks
 
-  // useEffect for row selection
   useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [hasSelectableRows]);
+    // We have to search the dom for the table ref because react-intl v2 does not support forwardRef.
 
-  // useEffect for row displayed columns
-  useEffect(() => {
-    setDataGridColumns(displayedColumns.map((column) => initializeColumn(column)));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [pinnedColumns, overflowColumns]);
-
-  // useEffect to calculate pinned column offsets
-  useEffect(() => {
-    const offsetArray = [];
-    let cumulativeOffset = 0;
-    let lastPinnedColumnIndex;
-
-    // if grid has selectable rows but no pinned columns, then set the offset of the first column to 0
-    if (hasSelectableRows && pinnedColumns.length === 0) {
-      lastPinnedColumnIndex = 0;
-      offsetArray.push(cumulativeOffset);
-      setPinnedColumnOffsets(offsetArray);
-      return;
-    }
-
-    if (pinnedColumns.length > 0) {
-      offsetArray.push(cumulativeOffset);
-
-      lastPinnedColumnIndex = hasSelectableRows ? pinnedColumns.length : pinnedColumns.length - 1;
-
-      // eslint-disable-next-line array-callback-return
-      dataGridColumns.slice(0, lastPinnedColumnIndex).map((pinnedColumn) => {
-        cumulativeOffset += pinnedColumn.width;
-        offsetArray.push(cumulativeOffset);
-      });
-    }
-    setPinnedColumnOffsets(offsetArray);
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [dataGridColumns]);
+    const [table] = gridContainerRef.current.getElementsByTagName('table');
+    grid.current = table;
+  }, [gridContainerRef]);
 
   // -------------------------------------
 
   const handleMoveCellFocus = (fromCell, toCell) => {
     // Obtain coordinate rectangles for grid container, column header, and new cell selection
-    const gridContainerRect = gridContainerRef.current.getBoundingClientRect();
+    const tableContainer = gridContainerRef.current.firstChild;
```

Comment:
Assuming that the table container is the first child would allow the logic to be easily broken.  If another element is added, it would break the implementation.  We probably need a better way to find the reference to this element.